hashring/hashring_test.go

```diff
@@ -52,42 +52,45 @@ func (d *dummyListener) HandleEvent(event events.Event) {
 	d.l.Unlock()
 }
 
-func TestEvents(t *testing.T) {
+func TestAddServer(t *testing.T) {
 	ring := New(farm.Fingerprint32, 10)
 	l := &dummyListener{}
 	ring.RegisterListener(l)
-	ring.emit(struct{}{})
-	assert.Equal(t, 1, l.events, "expected one event to be emitted")
-}
 
-func TestAddServer(t *testing.T) {
-	ring := New(farm.Fingerprint32, 10)
 	ring.AddServer("server1")
 	ring.AddServer("server2")
-
+	assert.Equal(t, 4, l.EventCount())
```

Comment:
I know it's been mentioned before—and this is just a minor thing—but I think we should move to using generated mocks where possible, for the sake of ease and standardisation.

With the generated mock you could do:

```
// Set up mock event listener with expectations
mockEventListener := &mocks.EventListener{}
mockEventListener.On("HandleEvent", mock.AnythingOfType("RingChangedEvent")).Times(5)
ring.RegisterListener(mockEventListener)

// Your test code here ...

// Assert expectations
mockEventListener.AssertExpectations()    // Will complain if the RingChangedEvent wan't fired 5 times
```

An example is here in this PR:
https://github.com/uber/ringpop-go/blob/tests/swim/swim/ping_test.go#L85

Note that I don't think this is a blocker for landing the PR. Just a nice-to-have.